### DISCUSSION TITLE
The form.value has outdated value in a field's "on change" action chain

### DISCUSSION THREAD

**Discussion by dannyshin76** [Budibase/budibase#8791] [Help]
My scenario is:
1. I have a form with dozens of fields
2. I want to send **form.value** to DB automatically when a user make changes in "budget" field (instead of forcing the user to click SUBMIT). So I use "On Change" action of the field for that. 
3. However, in the "On Change" action, **form.value** always shows the data before the change.
4. In the action chain, I tried to update the "budget" field value to "{{ Field Value }}" before using the **form.value**. But no change.

My question is:
1. How can I use updated **form.value** in a field's On Change action chain?
2. If it isn't possible, is there a way to listen on the form value change instead of field value change?

Thanks in advance!

**Comment by melohagan** [comment]
Hey @dannyshin76 

Real-time form submission is fairly unorthodox. If you want to save local changes automatically, you could maybe save them to [App state](https://docs.budibase.com/docs/app-state) and then when the submit button is clicked, it writes the app state to the table?
